pkg/reconciler/tenancy/workspace/workspace_reconcile_phase.go

```diff
@@ -127,9 +127,9 @@ func (r *phaseReconciler) reconcile(ctx context.Context, workspace *tenancyv1alp
 	return reconcileStatusContinue, nil
 }
 
-// updateTerminalConditionsAndPhase checks if the workspace is ready by checking conditions and sets the phase accordingly.
+// terminalConditionPhase checks if the workspace is ready by checking conditions and sets the phase accordingly.
 // It returns true if the phase was changed, false otherwise.
-func updateTerminalConditionsAndPhase(workspace *tenancyv1alpha1.Workspace) bool {
+func terminalConditionPhase(workspace *tenancyv1alpha1.Workspace) bool {
```

Comment:
not really what I meant. Either call it `terminalConditionPhase` and make it side-effect-free (i.e. returning the phase string, or `updateTerminalConditionPhase` (without the "And").

Reply:
Taking second one :) As reconcilers already playing a lot with pointers and editing existing objects - this just felt natural.